### DISCUSSION TITLE
[Q&A] Is it possible to use hostnames for remotes?

### DISCUSSION THREAD

**Discussion by ThomasDerrien** [glpi-project/glpi-agent#702] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hello,

I was wondering if it is possible to use hostnames instead of IP addresses or IP ranges when performing a remote inventory with GLPI Agent. Indeed, I'd like to inventory a computer that has a dynamic IP address but a consistent hostname, such as "computer1". Is there a way to configure the agent to inventory "computer1" and resolve its IP address using DNS?

**Comment by g-bougard** [comment, marked as answer]
Hello,

let me assume you're talking about this problem using ToolBox. With ToolBox, this is just not possible to use hostname as it relies on [Net::IP perl objects](https://metacpan.org/pod/Net::IP).

But this is possible if you use the `remote` option or if you configure the remote with [glpi-remote](https://glpi-agent.readthedocs.io/en/latest/man/glpi-remote.html) script. The related documentation is available here: https://glpi-agent.readthedocs.io/en/latest/tasks/remote-inventory.html

For ToolBox "netscan", it could be interesting to also support host target using hostname in the future.
